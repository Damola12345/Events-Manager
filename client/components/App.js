import React from 'react';
import { Switch, Route, Router } from 'react-router-dom';
import history from '../history';
import Home from './Home';
import SignUp from './SignUp';
import SignIn from './SignIn';
import Centers from './Centers';
import AddCenter from './AddCenter'

import '../static/scss/center-details.scss';


/**
 * @description creates a component that takes in all the routes as children
 * @returns {class} App
 */
class App extends React.Component {
  /**
   * 
   * @returns {component} router
   * @memberof App
   */
  render() {
    return (
      <Router history={history}>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/signup" component={SignUp} />
          <Route exact path="/signin" component={SignIn} />
          <Route exact path="/centers" component={Centers} />
          <Route exact path="/addcenter" component={AddCenter} />
        </Switch>
      </Router>
    )
  }

}

export default App;
