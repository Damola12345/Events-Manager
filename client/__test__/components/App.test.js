import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';

describe('Index', () => {
  const wrapper = shallow(<App />);
  it('successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

