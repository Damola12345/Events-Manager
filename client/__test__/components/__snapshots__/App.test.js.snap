// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index successfully rendered 1`] = `
<ConnectedRouter
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/register"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/events/add"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/events/:id/edit"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/events/:id"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/events"
    />
    <Connect(AdminRoute)
      component={[Function]}
      path="/centers/add"
    />
    <Connect(AdminRoute)
      component={[Function]}
      path="/centers/:id/edit"
    />
    <Route
      component={[Function]}
      path="/centers/:id"
    />
    <Route
      component={[Function]}
      path="/centers"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</ConnectedRouter>
`;
